packages/devtools_app/lib/src/screens/vm_developer/vm_field_display.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../primitives/utils.dart';
+import '../../shared/common_widgets.dart';
+import '../../shared/table.dart';
+import '../../shared/theme.dart';
+import 'vm_developer_common_widgets.dart';
+import 'vm_object_model.dart';
+import 'vm_service_private_extensions.dart';
+
+/// A widget for the object inspector historyViewport, displaying information
+/// related to field objects in the Dart VM.
+class VmFieldDisplay extends StatelessWidget {
+  const VmFieldDisplay({
+    required this.field,
+  });
+
+  final FieldObject field;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: Column(
+            children: [
+              FieldInfoWidget(
+                fieldDataRows: _fieldDataRows(field),
+              ),
+              Flexible(
+                child: ListView(
+                  children: [
+                    RetainingPathWidget(
+                      retainingPath: field.retainingPath,
+                      onExpanded: _onExpandRetainingPath,
+                    ),
+                    InboundReferencesWidget(
+                      inboundReferences: field.inboundReferences,
+                      onExpanded: _onExpandInboundRefs,
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
```

Comment:
this code is duplicated in vm_class_display.dart. Make a helper widget class that takes a VmObject and can be used for both instead of duplicating here

Reply:
Wrote VmObjectDisplayBasicLayout widget to reuse this code.